### ISSUE TITLE
RuntimeError: The number of elements exceeds the specified limit

### ISSUE THREAD

**Issue by munish0838** [docarray/docarray#1585]
Getting this error when indexing nested docList

**Comment by JoanFM** [comment]
Can you please provide the code snippet for us to reproduce?

**Comment by munish0838** [comment]
What I am trying to do here is store a document and sub-index it paragraph wise so I can later match either paragraphs or complete documents
This is the code I cam up with: 

`from docarray import BaseDoc
from docarray.index import HnswDocumentIndex
from docarray.typing import NdArray
import numpy as np
import os
from docarray import DocList
from typing import Optional

class Inner(BaseDoc):
    embedding: NdArray[128]
    text: str
    
class MyDoc(BaseDoc):
    user : str
    file_name : str
    projectId : str
    embedding: NdArray[128]
    text: Optional[str]
    inner: DocList[Inner]


db = HnswDocumentIndex[MyDoc](work_dir='personal/hnsw')


big_docs = DocList[MyDoc]()
for i in os.listdir("files"):
    pdf_text = get_pdf_text(f"files/{i}")

    docs = DocList[Inner](
        [Inner(embedding=np.random.rand(128), text= s) for s in pdf_text.split("\n")]
    )
    big_docs.append(MyDoc(embedding=np.random.rand(128), text = pdf_text,user ="24", projectId = "1", inner = docs, file_name = i))
    
db.index(big_docs)

`

**Comment by JoanFM** [comment]
Can you provide a more detailed stack trace?

**Comment by munish0838** [comment]
```
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
/tmp/ipykernel_115/3635111961.py in <module>
----> 1 db.index(big_docs)

~/.conda/envs/python3/lib/python3.9/site-packages/docarray/index/backends/hnswlib.py in index(self, docs, **kwargs)
    245         data_by_columns = self._get_col_value_dict(docs_validated)
    246 
--> 247         self._index(data_by_columns, docs_validated, **kwargs)
    248 
    249         self._send_docs_to_sqlite(docs_validated)

~/.conda/envs/python3/lib/python3.9/site-packages/docarray/index/backends/hnswlib.py in _index(self, column_to_data, docs_validated)
    211         docs_validated: Sequence[BaseDoc] = [],
    212     ):
--> 213         self._index_subindex(column_to_data)
    214 
    215         # not needed, we implement `index` directly

~/.conda/envs/python3/lib/python3.9/site-packages/docarray/index/abstract.py in _index_subindex(self, column_to_data)
   1092                     doc for doc_list in column_to_data[col_name] for doc in doc_list
   1093                 ]
-> 1094                 self._subindices[col_name].index(docs)
   1095                 column_to_data.pop(col_name, None)
   1096 

~/.conda/envs/python3/lib/python3.9/site-packages/docarray/index/backends/hnswlib.py in index(self, docs, **kwargs)
    245         data_by_columns = self._get_col_value_dict(docs_validated)
    246 
--> 247         self._index(data_by_columns, docs_validated, **kwargs)
    248 
    249         self._send_docs_to_sqlite(docs_validated)

~/.conda/envs/python3/lib/python3.9/site-packages/docarray/index/backends/hnswlib.py in _index(self, column_to_data, docs_validated)
    221             data_np = [self._to_numpy(arr) for arr in data]
    222             data_stacked = np.stack(data_np)
--> 223             index.add_items(data_stacked, ids=hashed_ids)
    224             index.save_index(self._hnsw_locations[col_name])
    225 

RuntimeError: The number of elements exceeds the specified limit
```

**Comment by JoanFM** [comment]
do u have a more dummy code to reproduce without needing access to the `folder`?

**Comment by munish0838** [comment]
I will write some dummy code and post it here soon

**Comment by munish0838** [comment]
Here it is, in the above code, replace the loop by following
```
pdf_text = '''Some text goes here \n line 2 \n line 3 \n line 4'''

docs = DocList[Inner](
    [Inner(embedding=np.random.rand(128), text= s) for s in pdf_text.split("\n")]
)
big_docs.append(MyDoc(embedding=np.random.rand(128), text = pdf_text,user ="24", projectId = "1", inner = docs, file_name = 0))

```

**Comment by JoanFM** [comment]
To me, this works properly:

```python
from docarray import BaseDoc
from docarray.index import HnswDocumentIndex
from docarray.typing import NdArray
import numpy as np
import os
from docarray import DocList
from typing import Optional

class Inner(BaseDoc):
    embedding: NdArray[128]
    text: str

class MyDoc(BaseDoc):
    user : str
    file_name : str
    projectId : str
    embedding: NdArray[128]
    text: Optional[str]
    inner: DocList[Inner]


db = HnswDocumentIndex[MyDoc](work_dir='./personal/hnsw')
big_docs = DocList[MyDoc]()
pdf_text = '''Some text goes here \n line 2 \n line 3 \n line 4'''

docs = DocList[Inner](
    [Inner(embedding=np.random.rand(128), text= s) for s in pdf_text.split("\n")]
)
big_docs.append(MyDoc(embedding=np.random.rand(128), text = pdf_text,user ="24", projectId = "1", inner = docs, file_name = 0))
```
I do not reproduce the issue, what exact issue do u have?

**Comment by munish0838** [comment]
Sorry missed the last line in above code
Try indexing `big_docs` in `db`, that is when it produces error
```db.index(big_docs)```

**Comment by JoanFM** [comment]
Still working for me:

```python
from docarray import BaseDoc
from docarray.index import HnswDocumentIndex
from docarray.typing import NdArray
import numpy as np
from docarray import DocList
from typing import Optional

class Inner(BaseDoc):
    embedding: NdArray[128]
    text: str
    
class MyDoc(BaseDoc):
    user : str
    file_name : str
    projectId : str
    embedding: NdArray[128]
    text: Optional[str]
    inner: DocList[Inner]


db = HnswDocumentIndex[MyDoc](work_dir='./personal/hnsw')
big_docs = DocList[MyDoc]()
pdf_text = '''Some text goes here \n line 2 \n line 3 \n line 4'''

docs = DocList[Inner](
    [Inner(embedding=np.random.rand(128), text= s) for s in pdf_text.split("\n")]
)
big_docs.append(MyDoc(embedding=np.random.rand(128), text = pdf_text,user ="24", projectId = "1", inner = docs, file_name = 0))
db.index(big_docs)
print(f'num_docs {db.num_docs()}')


```

**Comment by munish0838** [comment]
Is there a limit on the indexing?
Because in the folder I have 9 documents that are around 20-30 pages each 
When I try to index them, I get the error again

**Comment by JoanFM** [comment]
If u can share an example where I can reproduce the errorr, I can investigate. But no, there is no theoretical limit

**Comment by munish0838** [comment]
I will experiment a bit more and try to update you with an example or progress
Thanks a lot for help

**Comment by JoanFM** [comment]
I believe this has been fixed by #1602

**Comment by munish0838** [comment]
This fixed the issue
Sorry I couldn't help much with reproducing error or identifying the root cause
Thanks a lot
